healthcare/v1beta1/src/main/java/snippets/healthcare/fhir/resources/FhirResourceConditionalPatch.java

```diff
@@ -17,58 +17,83 @@
 package snippets.healthcare.fhir.resources;
 
 // [START healthcare_conditional_patch_resource]
+
 import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
 import com.google.api.client.http.HttpHeaders;
 import com.google.api.client.http.HttpRequestInitializer;
 import com.google.api.client.http.javanet.NetHttpTransport;
 import com.google.api.client.json.JsonFactory;
 import com.google.api.client.json.jackson2.JacksonFactory;
 import com.google.api.services.healthcare.v1beta1.CloudHealthcare;
-import com.google.api.services.healthcare.v1beta1.CloudHealthcare.Projects.Locations.Datasets.FhirStores.Fhir;
-import com.google.api.services.healthcare.v1beta1.CloudHealthcare.Projects.Locations.Datasets.FhirStores.Fhir.ConditionalPatchResource;
 import com.google.api.services.healthcare.v1beta1.CloudHealthcareScopes;
 import com.google.api.services.healthcare.v1beta1.model.HttpBody;
+
 import java.io.IOException;
+import java.net.URISyntaxException;
 import java.util.Collections;
 
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.HttpClients;
+
 public class FhirResourceConditionalPatch {
   private static final String FHIR_NAME =
       "projects/%s/locations/%s/datasets/%s/fhirStores/%s/fhir/%s";
+  private static final String FHIR_RESOURCE_NAME =
+      "projects/%s/locations/%s/datasets/%s/fhirStores/%s/fhir/%s/%s/%s";
   private static final JsonFactory JSON_FACTORY = new JacksonFactory();
   private static final NetHttpTransport HTTP_TRANSPORT = new NetHttpTransport();
 
-  public static void fhirResourceConditionalPatch(String resourceName) throws IOException {
+  public static void fhirResourceConditionalPatch(
+      String fhirStoreName,
+      String resourceType,
+      String data) throws IOException, URISyntaxException {
     // String resourceName =
     //    String.format(
     //        FHIR_NAME, "project-id", "region-id", "dataset-id", "store-id", "fhir-id");
-    String resourceType = "Patient";
+    // String resourceType = "Patient";
```

Comment:
Can we leave resourceType and data inside the function? For the [canonical](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/dlp/src/main/java/dlp/snippets/InspectImageFile.java#L52), the image type was deemed unnecessary to pass in through the test, so I moved it inline

Also for resourceType, is there a specific set of allowed strings? Or is it entirely up to the user? Might be nice to have a link explaining what the field could be